src/Elasticsearch.Net/Transport/Pipeline/RequestData.cs

```diff
@@ -113,23 +113,13 @@ private string CreatePathWithQueryStrings(string path, IConnectionConfigurationV
 			return path;
 		}
 
-
 		protected bool Equals(RequestData other) =>

```

Comment:
thought bubble, move GetHashCode() out of `RequestData` and into CoreFx's `HttpConnection` its cache is already `ConcurrentDictionary<int, HttpClient>` so we do not have to rely on `GetHashCode()` of `RequestData()`. 

Then we can get rid of the equals too and rely on reference comparison. Eventhough RequestData is immutable comparing two against eachother as values is not ever needed.